File: 0392-is-subsequence/0392-is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    let start_s = 0;
    let start_t = 0;
    
    if (s.length == 0){
        return true
    }
    
    while (start_t < t.length){
        if (s[start_s] === t[start_t]){
            start_s++;
        }
        
        if (start_s === s.length){
            return true
        }
        
        start_t++;
    }
    
    return false;
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0: return True
            
        start_s = 0;
        start_t = 0;

        while start_t < len(t):
            if s[start_s] == t[start_t]:
                start_s += 1
            if start_s == len(s):
                return True
            start_t += 1        

        return False